### PR TITLE
Fix typos in docs/reference/builder.md.

### PR DESCRIPTION
**Pull request by danix800**
Fix typos in docs/reference/builder.md.

Signed-off-by: Ding Fei dingfei@stars.org.cn

### CODE REVIEW

File: docs/reference/builder.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ performance, exclude files and directories by adding a `.dockerignore` file to
 the context directory.  For information about how to [create a `.dockerignore`
 file](#dockerignore-file) see the documentation on this page.
 
-Traditionally, the `Dockerfile` is called `Dockerfile` and located in the root
+Traditionally, the building file is called `Dockerfile` and located in the root
```
Comment (thaJeztah) [inline docs/reference/builder.md]: I'm a bit unsure if we should use "building file" here, as that would be introducing a new concept. I can see that the current sentence looks a bit funny, but it _is_ correct.

File: docs/reference/builder.md [hunk 2]
```diff
@@ -128,7 +128,7 @@ Docker runs instructions in a `Dockerfile` in order. **The first
 instruction must be \`FROM\`** in order to specify the [*Base
 Image*](glossary.md#base-image) from which you are building.
 
-Docker treats lines that *begin* with `#` as a comment, unless the line is
+Docker treats line that *begins* with `#` as a comment, unless the line is
```
Comment (thaJeztah) [inline docs/reference/builder.md]: The original line looks to be correct, can you revert this change?

File: docs/reference/builder.md [hunk 3]
```diff
@@ -519,14 +519,14 @@ command.
 
 In the *shell* form you can use a `\` (backslash) to continue a single
 RUN instruction onto the next line. For example, consider these two lines:
-```
-RUN /bin/bash -c 'source $HOME/.bashrc ;\
-echo $HOME'
-```
+
+    RUN /bin/bash -c 'source $HOME/.bashrc; \
```
Comment (thaJeztah) [inline docs/reference/builder.md]: Is there a reason you changed this (and removed the "fences")? We're using GitHub flavored markdown https://guides.github.com/features/mastering-markdown/

To improve this, you could add a `Dockerfile` code hint to improve syntax highlighting;


    ```Dockerfile
    RUN /bin/bash -c 'source $HOME/.bashrc ;\
    echo $HOME'
     ```

File: docs/reference/builder.md [hunk 4]
```diff
@@ -519,14 +519,14 @@ command.
...
+    echo $HOME'
+
 Together they are equivalent to this single line:
-```
-RUN /bin/bash -c 'source $HOME/.bashrc ; echo $HOME'
-```
+
+    RUN /bin/bash -c 'source $HOME/.bashrc; echo $HOME'
```
Comment (thaJeztah) [inline docs/reference/builder.md]: Same here

File: docs/reference/builder.md [hunk 5]
```diff
@@ -806,7 +806,7 @@ of whether or not the file has changed and the cache should be updated.
 > can only contain a URL based `ADD` instruction. You can also pass a
 > compressed archive through STDIN: (`docker build - < archive.tar.gz`),
 > the `Dockerfile` at the root of the archive and the rest of the
-> archive will get used at the context of the build.
+> archive will be used at the context of the build.
```
Comment (thaJeztah) [inline docs/reference/builder.md]: oh, looks like there's another typo here;

"used at the context" should be "used _as_ the context"

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks @danix800, I left some comments / suggestions inline

**Comment by danix800** [comment]
@thaJeztah Thanks for the review. Requested changes are squashed into the first commit, please take a look again.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
